docs/0.1/uploading_smart_contracts.md

```diff
@@ -0,0 +1,254 @@
+  in Docker containers.) This procedure assumes that there are two nodes,
+  `alpha-node-000` and `beta-node-000`, running in a Docker environment.
+
+* An approved Splinter circuit with two or more member nodes.
+  (See [Creating Splinter Circuits]({% link
+  docs/0.1/creating_splinter_circuits.md %}) for more information.)
+  This procedure assumes that there is a circuit with `alpha-node-000` and
+  `beta-node-000` as members.
+
+* A fully qualified service ID for the scabbard service on this circuit, in
+  the format <code><i>CircuitID</i>::<i>ServiceString</i></code>.
+  (See [Determine the Service ID]({% link docs/0.1/creating_splinter_circuits.md
+  %}#determine-the-service-id) for the commands to display this
+  information.) This procedure uses the example ID `01234-ABCDE::gsAA`.
+
+* The endpoint for the Splinter (`splinterd`) REST API. This example uses
+  `http://splinterd-alpha:8085`.
+
+* The public/private key pair for the smart contract owner. This example assumes
+  that Grid daemon (`gridd`) will be the smart contract owner and will sign
+  transactions with a private key in `/root/.splinter/keys/gridd.priv`.
+
+* A smart contract that has been compiled to WASM (as described in the
+  [Sabre Application Developer's
+  Guide](https://sawtooth.hyperledger.org/docs/sabre/releases/latest/application_developer_guide.html)).
+  This example uses the `sawtooth_xo` smart contract that is already packaged
+  in a smart contract archive (scar) file.
+
+## Important Notes
+
+The examples in this procedure use the node hostnames, container names, node
+IDs, and URLs that are defined in the example docker-compose file,
+`examples/splinter/docker-compose.yaml`. If you are not using this example
+environment, replace these items with the actual values for your nodes.
+
+## Procedure
+
+### Package the Smart Contract
+
+The examples in this procedure uses a pre-packaged smart contract,
+`sawtooth_xo`, which is available at
+[files.splinter.dev/scar/xo_0.4.2.scar](https://files.splinter.dev/scar/xo_0.4.2.scar).
+
+To use a different smart contract, follow these these steps to package the
+smart contract.
+
+1. Create a `manifest.yaml` file that specifies the smart contract name,
+   version, and the inputs and outputs (the addresses in state that the smart
+   contract will read from and write to).
+
+   The example smart contract has a manifest with the following contents:
+
+    ``` yaml
+    name: sawtooth_xo
+    version: '1.0'
+    inputs:
+      - '5b7349'
+    outputs:
+      - '5b7349'
+    ```
+
+1. Package the smart contract WASM file and the manifest in a smart contract
+   archive (a tar file with the `.scar` extension).
+
+    ``` console
+    $ tar -jcvf mycontract_0.1.0.scar mycontract.wasm manifest.yaml
+    ```
+
+    **IMPORTANT**: The name of the scar file must use the format
+    <code><i>name</i>_<i>version</i></code> (for example, `mycontract_0.1.0`).
```

Comment:
This should note that the version must be valid semver version (and provide a link to semver.org).